### ISSUE TITLE
1.0.0.alpha.17: transition with & without delay outputted incorrectly for webkit

### ISSUE THREAD

**Issue by thany** [Compass/compass#1550]
Consider this sass:

```
@include transition(.2s ease-in opacity, .2s .2s ease-in visibility);
```

This will output css that is incorrect only for webkit:

```
-moz-transition: 0.2s ease-in opacity, 0.2s 0.2s ease-in visibility;
-o-transition: 0.2s ease-in opacity, 0.2s 0.2s ease-in visibility;
-webkit-transition: 0.2s ease-in opacity, 0.2s 0.2s ease-in;
-webkit-transition-delay: 0s, visibility;
transition: 0.2s ease-in opacity, 0.2s 0.2s ease-in visibility;
```

As you can see, other prefixes are outputted correctly.

Related testcase: http://sassmeister.com/gist/8861702

**Comment by kroleg** [comment]
Hi guys. 
I have a fix for this https://github.com/kroleg/compass/commit/e7dbe9e0690ee96dd06eee16ca256efbaf16005c
But: 
1) It's incomplete (i don't how how to check if one of values passed to transition mixin is "cubic-bezier()")
2) Regression test doesn't pass. What is it for?
3) Contribution policy is a bit confusing for me.
Can you please give advice me, how to solve (1) and (2)? And what i need to do before a pull request?
Thanks!

**Comment by eddie-ruva** [comment]
@kroleg were you able to fix your problems with (1) and (2)? I would like to take a look at how we can fix this but before starting to code I wanted to check with you if you were able to get something good working so we don't have to start from scratch. 

By taking a look at the code I think a quick solution for anyone that runs into the problem before a fix is submitted is to call the mixin in the following way:

``` scss
// transition: <property> <duration> <timing-function> <delay>
@include transition(opacity .2s ease-in, visibility .2s ease-in .2s );
```

**Comment by eddie-ruva** [comment]
[Compass docs for css3 transitions](http://compass-style.org/reference/compass/css3/transition/) mention that the order of the parameters for creating a transition are: 

``` scss
@include single-transition($property, $duration, [$function], [$delay])
@include transition([$transition-1], [$transition-2], [$transition-3], [$transition-4], [$transition-5], [$transition-6], [$transition-7], [$transition-8], [$transition-9], [$transition-10])
```

Also the code in the transition mixin seems to be very couple to the order in which the parameters are passed:

``` scss
$property: nth($transition, 1);
$duration: if(length($transition) >= 2, nth($transition, 2), null);
$timing-function: if(length($transition) >= 3, nth($transition, 3), null);
$delay: if(length($transition) >= 4, nth($transition, 4), null);
```

Having said that there can probably be two changes that can be done: 
1. Modify the transition mixin so that it knows how the handle the parameters even when they are not provided in the desired order which is what kroleg@e7dbe9e is trying to do.
2. Maybe make a change in the docs and be more explicit about the order in which the parameters are expected and keep the code as it is.

cc/ @chriseppstein

**Comment by thany** [comment]
To keep in check with the W3C recommended notation, I suggest changing the mixin behaviour so it can work with parameters in any order. This should go for other mixins as well, of course, such as box-shadow (may not be neccesary though).

The CSS3 spec prescribes how parameters must be interpreted. What's more (if I remember) the spec never dictated the order of parameters. Just which is which when for example two lengths may appear.

**Comment by chriseppstein** [comment]
Fixed by f296478c29518f3d02786c5fd62fdb0ccca8ad80
